app/controllers/admin/agent_agendas_controller.rb

```diff
@@ -9,4 +9,17 @@ def show
     @selected_event_id = params[:selected_event_id]
     @date = params[:date].present? ? Date.parse(params[:date]) : nil
   end
+
+  def toggle_display_saturdays
+    @agent = current_agent
+    authorize(@agent)
+    @agent.update(agent_role_params)
```

Comment:
```suggestion
    @agent.update!(agent_role_params)
```
bang bang bang?
bang alors ?